core/src/test/java/io/github/tcdl/msb/message/MessageFactoryTest.java

```diff
@@ -219,4 +220,28 @@ public void testCreateAckBuilder() throws Exception {
         Acknowledge ack = messageFactory.createAckBuilder().build();
         assertNotNull(ack.getResponderId());
     }
+
+    @Test
+    public void testCreateRequestMessageBuilderPublishedAtPresent() {
+        String bodyText = "body text";
+        Payload requestPayload = TestUtils.createPayloadWithTextBody(bodyText);
+
+        Builder requestMessageBuilder = TestUtils.createMessageBuilder();
+
+        Message message = messageFactory.createRequestMessage(requestMessageBuilder, requestPayload);
+
+        assertNotNull(message.getMeta().getPublishedAt());
+    }
+
+    @Test
+    public void testCreateRequestMessageBuilderPublishedAtIsAfterCreatedAt() {
+        String bodyText = "body text";
+        Payload requestPayload = TestUtils.createPayloadWithTextBody(bodyText);
+
+        Builder requestMessageBuilder = TestUtils.createMessageBuilder();
+
+        Message message = messageFactory.createRequestMessage(requestMessageBuilder, requestPayload);
+
+        assertTrue(message.getMeta().getPublishedAt().isAfter(message.getMeta().getCreatedAt()));
```

Comment:
Also I would suggest creating stronger test by using response messages when you created a responder and send 2 messages (ack and actual response). In such test you could check that that all 3 dates (createdAt and 2 publishedAt are one after another)